### DISCUSSION TITLE
Force ignore the Godot version?

### DISCUSSION THREAD

**Discussion by tiloc** [godot-jolt/godot-jolt#841] [Questions] [answered]
I am more and more looking forward to switching my project to the upcoming Godot 4.3, but Jolt is steadfastly insisting it can only work with 4.2. Is there a file where I can loosen that restriction at my own risk? Is there a reasonable expectation that this might work?

**Comment by mihe** [comment, marked as answer]
> Is there a file where I can loosen that restriction at my own risk?

There is not, no. That version check is currently [hardcoded](https://github.com/godot-jolt/godot-jolt/blob/aeeaee05f81eb7b73bebf52ac08c5635f3f93941/cmake/GodotJoltVersionInfo.cmake#L16), so you would need to change that and build from source.

In the next version of Godot Jolt (which will target Godot 4.3) you'll be able to modify a `compatibility_maximum` version number in the `.gdextension` file instead.

> Is there a reasonable expectation that this might work?

Godot itself aims for extensions to be forward-compatible, but historically this has been rocky, which is why the version check is there. When binary incompatibilities creep in things can break in subtle (and not so subtle) ways.

I'm not aware of any such breakages in 4.3, but since the version check saves me from caring about them I haven't really stayed up-to-date on it either.

Anyway, if you don't feel like building from source and all that stuff then there will be a pre-release here on GitHub some time around 4.3-beta1.
